worldedit-core/src/main/java/com/sk89q/worldedit/LocalSession.java

```diff
@@ -559,34 +561,40 @@ public boolean toggleSuperPickAxe() {
      * @throws IncompleteRegionException thrown if a region is not fully selected
      */
     public BlockVector3 getPlacementPosition(Actor actor) throws IncompleteRegionException {
-        checkNotNull(actor);
-        if (!placeAtPos1) {
-            if (actor instanceof Locatable) {
-                return ((Locatable) actor).getBlockLocation().toVector().toBlockPoint();
-            } else {
-                throw new IncompleteRegionException();
-            }
-        }
+        return this.placement.getPlacementPosition(selector, actor);
+    }
+
+    public Placement getPlacement() {
+        return this.placement;
+    }
 
-        return selector.getPrimaryPosition();
+    public void setPlacement(Placement placement) {
+        this.placement = placement;
     }
 
+    @Deprecated
```

Comment:
could we have a javadoc `@deprecated` tag here explaining what to use instead so people know where to migrate

Reply:
Will be added in my next push